File: streaming_yorkie/src/main/java/com/lethalmaus/streaming_yorkie/entity/FollowerEntity.java
package com.lethalmaus.streaming_yorkie.entity;

import androidx.room.Entity;

/**
 * FollowerEntity Entity that extends UserEntity Entity
 * @author LethalMaus
 */
@Entity(tableName = "follower")
public class FollowerEntity extends UserEntity {

    /**
     * FollowerEntity Constructor
     * @param id Twitch Id
     * @param display_name Twitch Name
     * @param logo Twitch Logo
     * @param created_at FollowingEntity Since
     * @param notifications Notifications Activated
     * @param last_updated Last retrieved from Twitch
     */
    public FollowerEntity(int id, String display_name, String logo, String created_at, boolean notifications, long last_updated) {
        super(id, display_name, logo, created_at, notifications, last_updated);
    }
}

File: app/src/main/java/com/lethalmaus/streaming_yorkie/database/entity/FollowerEntity.kt
package com.lethalmaus.streaming_yorkie.database.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "follower")
class FollowerEntity (
    @field:PrimaryKey
    override var id: Int,
    @ColumnInfo(name = "display_name")
    override var displayName: String?,
    override var logo: String? = null,
    @ColumnInfo(name = "created_at")
    override var createdAt: String?,
    override var notifications: Boolean,
    @ColumnInfo(name = "last_updated")
    override var lastUpdated: Long
) : UserEntity(id, displayName, logo, createdAt, notifications, lastUpdated)